tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/expressions/builtinfunctions/FreezeAndIsFrozenTest.java

```diff
@@ -442,58 +442,54 @@ public void testXMLItemsCloneReadOnly() {
 
     @Test
     public void testFreezeAndIsFrozenSemanticsNegativeCases() {
-        Assert.assertEquals(semanticsNegativeResult.getErrorCount(), 24);
+        Assert.assertEquals(semanticsNegativeResult.getErrorCount(), 22);
         int index = 0;
         validateError(semanticsNegativeResult, index++,
-                "incompatible types: expected 'PersonObj', found 'anydata'", 19, 19);
+                "incompatible types: expected 'CloneableType', found 'PersonObj'", 19, 19);
```

Comment:
Shouldn't the errors be like
```suggestion
                "incompatible types: expected 'Cloneable', found 'PersonObj'", 19, 19);
```

Reply:
There are three errors for the same line (line 19). One is out of order now.
1. PersonObj -> anydata
2. anydata -> PersonObj
3. anydata -> PersonObj


now 
2. CloneableType -> PersonObj
3. CloneableType -> PersonObj
1. PersonObj -> CloneableType

Reply:
Sorry, what I meant was `Cloneable` instead of `CloneableType`. `CloneableType` is the type param definition that is not public, right?

Reply:
The problem is to prevent TYPE param flag propagation. I have defined `CloneableType` as `Cloneable|never` as a workaround for now. Therefore type param is taking its own name. 

Issue tracking never introduction
https://github.com/ballerina-platform/ballerina-lang/issues/27786

Reply:
Because we print the name for unions now if it has a name? Thinking out loud, wonder if we should special case this until we fix #27786.

The error message will be confusing if not, because the user wouldn't know what `CloneableType` is?

Reply:
Only for cyclic unions. I think I can add a logic there to check if the type is public or private.

Reply:
Then it looks like "incompatible types: expected '(readonly|xml|Cloneable[]|map<Cloneable>|table<map<Cloneable>>)', found 'PersonObj'".

This is how `CloneableType` is created now
<img width="420" alt="Screenshot 2021-01-19 at 12 22 13 PM" src="https://user-images.githubusercontent.com/2173530/104998337-08b71880-5a51-11eb-881b-d58838066355.png">

Reply:
Added a hack by commit
https://github.com/ballerina-platform/ballerina-lang/pull/27909/commits/3260ab1e295eb50e07331dad9a0110d1310e76d5